sdk/file_sdk/Common/file_format_base.h

```diff
@@ -0,0 +1,36 @@
+#ifndef FILE_FORMAT_BASE_H
+#define FILE_FORMAT_BASE_H
+
+#include <string>
+#include <vector>
+#include "../../rmscrypto_sdk/CryptoAPI/IStream.h"
+#include "tag.h"
+using std::string;
+using std::vector;
+
+namespace mip{
+namespace file {
+
+class IFileFormat
+{
+public:
+  IFileFormat(rmscrypto::api::SharedStream& file, string extension);
+
+  virtual string GetOriginalExtension() = 0;
+
+  virtual void SetTags(vector<Tag>& tags);

```

Comment:
ok, although i disagree.  i checked existing "interfaces" and the classes are not abstract(like IStream), i don't think we need another class for 2 lines just because this class name starts with "I"


---
In reply to: [134386959](https://github.com/AzureAD/rms-sdk-for-cpp/pull/133#discussion_r134386959) [](ancestors = 134386959,134246608)